File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
 
public class Solution {
    public List<List<int>> LevelOrder(TreeNode root) {
        // folgende Idee: wir mahcen das mit breitensuche
        // jedes mall wenn wir durchegehen duch eine tiefe isnatnz 
        // poppen wir das einfach in in ne lsute und fertgi 
        List<List<int>> res = new List<List<int>>();
        if(root == null) return res;
        Queue<TreeNode> buffer = new Queue<TreeNode>();
        buffer.Enqueue(root); 
        while (buffer.Count > 0)
        {
        List<int> cont = new List <int>();
        int threshold = buffer.Count;
        for(int i = 0; i < threshold; i++)
        {
            TreeNode node = buffer.Dequeue();
            if(node != null)
            {
            cont.Add(node.val);   
            if(node.left != null) buffer.Enqueue(node.left);
            if(node.right != null) buffer.Enqueue(node.right);
            }
        }
        if(cont.Count > 0)
        {
            res.Add(cont);
        }


        }
        return res; 


    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        
        
        
        res = []

        q = deque()
        q.append(root)
        while q: 
            lenq = len(q)
            for i in range(lenq):
                nodes = q.popleft()
                
                if nodes.left:
                    q.append(nodes.left)
                
                if nodes.right:
                    q.append(nodes.right)

                if  i == lenq-1:
                    res.append(nodes.val)
            
        return res